test/torchaudio_unittest/datasets/tedlium_test.py

```diff
@@ -151,3 +149,11 @@ def test_tedlium_release3(self):
         phoneme_dict = dataset.phoneme_dict
         phoenemes = [f"{key} {' '.join(value)}" for key, value in phoneme_dict.items()]
         assert phoenemes == PHONEME
+
+
+class TestTedliumDefault(Tedlium):
+    backend = "default"
+
+
+class TestTedliumSoundfile(Tedlium):
+    backend = "soundfile"
```

Comment:
This means that sound file test is ran twice on Windows machine.
It would be better if this one is excluded in Windows.

Reply:
Also, it is correctly failing on Windows CPU test. I do not think it is necessary to add this test for linux.

https://app.circleci.com/pipelines/github/pytorch/audio/3479/workflows/0f7c62e4-e357-4c20-bab7-98b165b1a8f4/jobs/101683

Reply:
Actually, `backend = "soundfile"` is not enough.
Instead of adding the default `"soundfile"` backend, we need to introduce `"soundfile-new"` backend value to `TorchaudioTestCase` and make sure that the `"soundfile"`'s new interface is tested.

Reply:
> Also, it is correctly failing on Windows CPU test. I do not think it is necessary to add this test for linux.

I agree that Windows test would cover it. Based on [comment](https://github.com/pytorch/audio/pull/934#pullrequestreview-501264622), I'll leave it in though. The argument I see in favor is that we may change the default backend on Windows at some and we want to make sure this code path is covered.

Reply:
> Actually, `backend = "soundfile"` is not enough.
> Instead of adding the default `"soundfile"` backend, we need to introduce `"soundfile-new"` backend value to `TorchaudioTestCase` and make sure that the `"soundfile"`'s new interface is tested.

Added a `soundfile-new` option for `TorchaudioTestCase`. I do force legacy backend due to the global variable that would be changed by the soundfile-new backend.

Reply:
I also realized that `"sox_io"` backend is missing from the test too.

Can you remove `"default"` , then add `"soundfile"`, `"soundfile-new"` and `"sox"`, `"sox_io"` for non-Windows?

```
class TestTedliumSoundfile(Tedlium, TorchaudioTestCase):
     backend = "soundfile"


class TestTedliumSoundfileNew(Tedlium, TorchaudioTestCase):
     backend = "soundfile-new"


if platform.system() != "Windows":
    class TestTedliumSox(Tedlium, TorchaudioTestCase):
         backend = "sox"


    class TestTedliumSoxIO(Tedlium, TorchaudioTestCase):
         backend = "sox_io"
```